chain/bitcoind_client.go

```diff
@@ -427,10 +427,44 @@ func (c *BitcoindClient) RescanBlocks(
 	blockHashes []chainhash.Hash) ([]btcjson.RescannedBlock, error) {
 
 	rescannedBlocks := make([]btcjson.RescannedBlock, 0, len(blockHashes))
+
+	// requests map will store rpcclient.FutureGetBlockHeaderVerboseResult for
+	// each blockHash.
+	requests := make(
+		map[chainhash.Hash]rpcclient.FutureGetBlockHeaderVerboseResult,
+		len(blockHashes),
+	)
+
+	for _, hash := range blockHashes {
+		hash := hash
+
+		headerRequest, err := c.GetBlockHeaderVerboseAsync(&hash)
+		if err != nil {
+			log.Warnf(
+				"Unable to receive header promise from hash %v: %s", hash, err)
+			continue
+		}
+		requests[hash] = headerRequest
+	}
+
+	c.SendAsyncQueue()
```

Comment:
I think this function could be reworked a bit from its current form, to:

1) fetch block headers to get timestamps and heights (send async and receive too)
2) in the receive loop you could filter out those which are not important for us:
```
if !c.shouldFilterBlock(time.Unix(header.Time, 0)) {
			continue
}
```
3) finally get the filtered block hashes